### ISSUE TITLE
Function which return total paid for certain order using precision by context instead of order

### ISSUE THREAD

**Issue by integritty** [PrestaShop/PrestaShop#29124]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

When you have active multistore instance then function getTotalPaid in /classes/order/Order.php using current active context  (precision from current currency context). This causes incorrect behaviour and showing error which doesn't really exist.

In multistore context (with TWO decimails in precision)
![Screenshot 2022-07-21 at 13 58 58](https://user-images.githubusercontent.com/71325559/180208958-440f6eeb-fd2e-4248-b486-3902bbc03db4.png)

In multistore context (with ZERO decimails in precision)
![Screenshot 2022-07-21 at 13 58 49](https://user-images.githubusercontent.com/71325559/180208963-d53f74a5-ad9f-49fa-ab18-b19678811bd1.png)

### Expected behavior

Correct price validation on order should be calculated by id_currency inside current order object.

### Steps to reproduce

1. Create multistore instance of PrestaShop (CZ and COM, two stores)
2. Set two currencies EUR and CZK
3. For CZK set precision to 0 and for EUR set precision to CZK
4. Set currecy separately for CZ = CZK and COM = EUR
5. Place order for CZK and for EUR currency (EUR currency must have two decimails price on product)
6. Show order from admin area with context of currency with zero precision (CZK)
7. Warning is showing up above payment block

### PrestaShop version(s) where the bug happened

1.7.8.2

### PHP version(s) where the bug happened

7.4.30

**Comment by marwachelly** [comment]
Hello @integritty,

Could you please explain more the step 3 "For CZK set precision to 0 and for EUR set precision to CZK" 
Could you please provide a screen record to help as to reproduce the issue?
waiting for your feedback!

Many thanks,

**Comment by integritty** [comment]
@marwachelly Go to back office, then International -> Localization -> Currencies, there add CZK and EUR currency and set for CZK currency „Decimails“ to 0 (zero) and for EUR set „Decimals“ to 2 (two).
![Screenshot 2022-07-25 at 13 24 34](https://user-images.githubusercontent.com/71325559/180766915-6ed7f419-317d-4115-8ffd-3547d4fe531a.png)
![Screenshot 2022-07-25 at 13 24 31](https://user-images.githubusercontent.com/71325559/180766921-95b29ca7-db34-4119-ac1c-96444ae9969a.png)
![Screenshot 2022-07-25 at 13 24 25](https://user-images.githubusercontent.com/71325559/180766927-9e6b0ef4-e169-4779-850c-4b9d81e6a00c.png)

**Comment by florine2623** [comment]
Hello @integritty ,

Thanks for the details !

I'm using PS 1.7.8.2, PHP 7.4. On my shop I have 2 groups with 1 shop in each group.
I have followed your issue, but I can't reproduce the same error as you.
Here's my order from CZ shop, with CZK currency:
<img width="816" alt="Screenshot 2022-07-26 at 16 17 34" src="https://user-images.githubusercontent.com/16019289/181042383-755a5212-8838-464e-8220-3b97815e3297.png">

The paid amount is correct. I tested with multiple Order Status.

Is there more details to be tested ?
Thanks!
